src/main/java/seedu/clinic/logic/commands/UpdateCommand.java

```diff
@@ -129,12 +129,14 @@ private CommandResult updateProductForWarehouse(Model model) throws CommandExcep
     private CommandResult updateProductForSupplier(Model model) throws CommandException {
         Supplier supplierToUpdate;
 
-        try {
-            supplierToUpdate = getSupplierByName(entityName, model);
-        } catch (NoSuchElementException e) {
+        Optional<Supplier> optionalSupplier = model.getSupplier(entityName);
+
+        if (optionalSupplier.isEmpty()) {
             throw new CommandException(MESSAGE_NO_SUCH_ENTITY);
         }
 
+        supplierToUpdate = optionalSupplier.orElseThrow();
```

Comment:
Should the orElseThrow() have some exceptions as argument here?

Reply:
I think can just leave as default, because the expected commandexception will already be caught in the above in the optional check